File: solutions/3033. Modify the Matrix/3033.cpp
// Time complexity: O(mn)
// Space complexity: O(1)
class Solution {
public:
  vector<vector<int>> modifiedMatrix(vector<vector<int>>& matrix) {
    int rows = matrix.size();
    int cols = matrix[0].size();

    for (int i = 0; i < cols; ++i) {
      int maxVal = 0;
      for (int j = 0; j < rows; ++j)
        maxVal = max(maxVal, matrix[j][i]);

      for (int j = 0; j < rows; ++j) {
        if (matrix[j][i] == -1)
          matrix[j][i] = maxVal;
      }
    }

    return matrix;
  }
};

File: solutions/3033. Modify the Matrix/3033.py
# Time complexity: O(mn)
# Space complexity: O(1)
class Solution:
  def modifiedMatrix(self, matrix: List[List[int]]) -> List[List[int]]:
    rows = len(matrix)
    cols = len(matrix[0])

    for i in range(cols):
      maxVal = 0
      for j in range(rows):
        maxVal = max(maxVal, matrix[j][i])

      for j in range(rows):
        if matrix[j][i] == -1:
          matrix[j][i] = maxVal

    return matrix